spec/unit/shoulda/matchers/active_record/have_rich_text_matcher_spec.rb

```diff
@@ -0,0 +1,84 @@
+require 'unit_spec_helper'
+
+describe Shoulda::Matchers::ActiveRecord::HaveRichText, type: :model do
+  def self.rich_text_is_defined?
+    defined?(ActionText::RichText)
+  end
+
+  if rich_text_is_defined?
+    context 'when the model has a RichText association' do
+      it 'matches when the subject configures has_rich_text' do
+        valid_record = new_post(is_rich_text_association: true)
+
+        expected_message = 'Did not expect Post to have ActionText::RichText :content'
+
+        expect { have_rich_text(:content) }.
+          to match_against(valid_record).
+          or_fail_with(expected_message)
+      end
+
+      it 'returns the message including the name of the provided association' do
+        matcher = have_rich_text(:content)
+        expect(matcher.description).
+          to eq('have configured :content as a ActionText::RichText association')
+      end
+    end
+
+    context 'when the model does not have a RichText association' do
+      it 'does not match when provided with a model attribute' do
+        invalid_record = new_post(has_invalid_content: true)
+        expected_message = 'Expected Post to have configured :invalid_content as a ' \
```

Comment:
Okay, getting closer. Now that I see this message in its full form, I'm now wondering if it's is as helpful as it could be. For instance, `invalid_content` is defined a string column, but happens to be unset at the time this test is run. Does that information help the user realize what they need to change? I guess what I'm getting at is, would it be enough to say "Expected Post to have configured :invalid_content as a ActionText::RichText association" and not include the type at the end?

Reply:
Hehe 😄 so I might have miscommunicated here. What I meant was that I felt like only _this_ particular message should be changed. The other one that says ":invalid_content does not exist", I think that one makes sense to keep. It's important for people to know that an attribute doesn't exist (we do that for other matches as well), but perhaps not so important to know what the current value of the attribute is. So I think it still makes sense for lines 42 and 63 to say "Expected Post to have configured :invalid_attribute as an ActionText::RichText association, but :invalid_attribute does not exist".

Reply:
No you did not.

The problem was, I didn't know how to refactor this:

```
def run_checks
   if !has_attribute?
     ":#{rich_text_attribute} does not exist"
  elsif !has_expected_action_text?
    ":#{rich_text_attribute} is a #{@subject.send(rich_text_attribute).class}"
  end
end
```

The first question that pop up on my mind was: what will be returned by the second condition now? Because it does not need the description in the string. So maybe a empty string? I don't like that.

So in the end I removed the conditional and decided to make all the code more clean, but I you're right again. It's important to know that an attribute doesn't exist. 

I'll try again.

Thanks!

Reply:
Done!

What do you think?

Reply:
I like your solution! It's concise and yet understandable.